scalahost/src/main/scala/meta/Mirror.scala

```diff
@@ -5,9 +5,13 @@ import scala.collection.immutable.Seq
 import scala.tools.nsc.Global
 import scala.meta.internal.scalahost.v1.online.{Mirror => OnlineMirror}
 import scala.meta.internal.scalahost.v1.offline.{Mirror => OfflineMirror}
+import scala.util.Try
 
 object Mirror {
   def apply(global: Global): Mirror = new OnlineMirror(global)
   def apply(classpath: String, sourcepath: String): Mirror =
     new OfflineMirror(classpath, sourcepath)
+  def loadFromSysProps(): Try[Mirror] = Try {
+    apply(sys.props("scalameta.mirror.classpath"), sys.props("scalameta.mirror.sourcepath"))
+  }
```

Comment:
I suggest we change the signature to `def apply(): Mirror`. Here's the reasoning:

1) I'd say that `Try` (or `Option`, or `Either`) is useful when the error condition is non-fatal. Can the user do anything about this error? Can they do something alternative to achieve the same result? In our case, both answers are "no", so I'd advocate for using exceptions here.

2) I think that it is very important to have a hassle-free way to instantiate a mirror. How we implement it - via sbt properties, via analyzing `getClass.getClassLoader` - it doesn't matter that much imo. I suggest that we try to make this work via `Mirror()`, and if magic is confusing to the users then we drop it later. I guess we also have to provide a better exception message, e.g. `auto mirror is not configured. use sbt's ``% Scalameta`` to blah`.

/cc @densh